### PR TITLE
Added new Jan 20 Bronze Q2 Python Solution

### PR DESCRIPTION
**Pull request by AAkashyap1**

### CODE REVIEW

**Comment by ryanchou-dev** [comment]
did you forget to add it to extraproblems.json?

File: solutions/usaco-988.mdx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_photo_bronze_jan20.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<JavaSection>
```
Comment (ryanchou-dev) [inline solutions/usaco-988.mdx:16]:
```suggestion
<PySection>
```

File: solutions/usaco-988.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+for i in range(n - 1):
+  ans += (str(a[i]) + ' ')
+
+ans += str(a[n - 1])
+
+print(ans, file=open('photo.out', 'w'))
+```
+
+</JavaSection>
```
Comment (ryanchou-dev) [inline solutions/usaco-988.mdx:48]:
```suggestion
</PySection>
```

File: solutions/usaco-988.mdx (hunk 1 above)
Comment (ryanchou-dev) [reply solutions/usaco-988.mdx:16]: wrong language section?

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
Some minor fixes here:

**Comment by ryanchou-dev** [comment]
also, this already has a solutions page: [](https://usaco.guide/problems/usaco-989-race/solution)

should add your implementation there instead of creating a new one.

**Comment by AAkashyap1** [comment]
> also, this already has a solutions page: https://usaco.guide/problems/usaco-989-race/solution
> 
> should add your implementation there instead of creating a new one.

this is actually 988 which doesnt have a solutions page sorry my bad

**Comment by AAkashyap1** [comment]
> > also, this already has a solutions page: https://usaco.guide/problems/usaco-989-race/solution
> > should add your implementation there instead of creating a new one.
> 
> this is actually 988 which doesnt have a solutions page sorry my bad

i forgot to change the title

File: solutions/usaco-988.mdx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	a.append(i)
+	works = True
+	# Check if a valid permutation can be formed given the first number
+	for j in range(n - 1):
+		if ((b[j] - a[j]) in a) or ((b[j] - a[j]) == 0) or ((b[j] - a[j]) > n):
+			works = False
+			break
+		else:
+			a.append(b[j] - a[j])
+	if works == True:
```
Comment (ryanchou-dev) [inline solutions/usaco-988.mdx:35]:
```suggestion
	if works:
```

**Review by ryanchou-dev — COMMENTED** [review]
just this, rest looks good :)

File: solutions/usaco-988.mdx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+source: USACO Bronze 2020 January
+title: Photoshoot
+author: Ananth Kashyap
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_photo_bronze_jan20.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^2)$
```
Comment (dongliu0426) [inline solutions/usaco-988.mdx:12]: if I'm not wrong your implementation is O(N^3)
> if ((b[j] - a[j]) in a)

^^ this takes O(N) and you have two nested loops

Comment (dongliu0426) [reply solutions/usaco-988.mdx:12]: so please change it to O(N^2)

File: solutions/usaco-988.mdx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+<LanguageSection>
+
+<PySection>
+
+```py
+data = open('photo.in').read().strip().split('\n')
+n = int(data[0])
+b = list(map(int, data[1].split(' ')))
+
+a = []
```
Comment (dongliu0426) [inline solutions/usaco-988.mdx:23]: more descriptive variable names :)

File: solutions/usaco-988.mdx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+<LanguageSection>
+
+<PySection>
+
+```py
+data = open('photo.in').read().strip().split('\n')
+n = int(data[0])
+b = list(map(int, data[1].split(' ')))
```
Comment (dongliu0426) [inline solutions/usaco-988.mdx:21]: more descriptive variable names :)

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

**Review by ryanchou-dev — APPROVED** [review]
👍

File: solutions/usaco-988.mdx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+	if works:
+		break
+
+ans = ''
+
+for i in range(n - 1):
+	ans += (str(a[i]) + ' ')
+
+ans += str(a[n - 1])
+
```
Comment (SansPapyrus683) [inline solutions/usaco-988.mdx:44]: 39, 42, and 44 are unnecessary

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
minor

File: solutions/usaco-988.mdx (hunk 4 above)
Comment (jessechoe10) [reply solutions/usaco-988.mdx:12]:
```suggestion
**Time Complexity:** $\mathcal{O}(N^3)$
```

File: solutions/usaco-988.mdx (hunk 6 above)
Comment (jessechoe10) [reply solutions/usaco-988.mdx:21]: I'm pretty sure these var names are from the problem, so it should be relatively straightforwards as to what the prob is saying... 

Read this [comment](https://github.com/cpinitiative/usaco-guide/pull/1973#discussion_r675775588) for a more detailed reason.

File: solutions/usaco-988.mdx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+		if ((b[j] - a[j]) in a) or ((b[j] - a[j]) == 0) or ((b[j] - a[j]) > n):
+			works = False
+			break
+		else:
+			a.append(b[j] - a[j])
+	if works:
+		break
+
+ans = ''
+
```
Comment (jessechoe10) [inline solutions/usaco-988.mdx:39]:
```suggestion
```

File: solutions/usaco-988.mdx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+for i in range(n - 1):
+	ans += (str(a[i]) + ' ')
+
```
Comment (jessechoe10) [inline solutions/usaco-988.mdx:42]:
```suggestion
```

File: solutions/usaco-988.mdx (hunk 7 above)
Comment (jessechoe10) [reply solutions/usaco-988.mdx:44]:
```suggestion
```

**Review by jessechoe10 — CHANGES_REQUESTED** [review]

File: solutions/usaco-988.mdx (hunk 4 above)
Comment (dongliu0426) [reply solutions/usaco-988.mdx:12]: @jessechoe10 why?

File: solutions/usaco-988.mdx (hunk 6 above)
Comment (dongliu0426) [reply solutions/usaco-988.mdx:21]: k then

File: solutions/usaco-988.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/usaco-988.mdx:23]: you can keep the `n` but at least rename `a` and `b` they aren't explicit at all

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-988.mdx (hunk 5 above)
Comment (jessechoe10) [reply solutions/usaco-988.mdx:23]: they're technically from the problem :stuck_out_tongue:

**Review by jessechoe10 — APPROVED** [review]

File: solutions/usaco-988.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/usaco-988.mdx:23]: don't think anyone reads the series, please just change those

File: solutions/usaco-988.mdx [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+source: USACO Bronze 2020 January
+title: Photoshoot
+author: Ananth Kashyap
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_photo_bronze_jan20.html)
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N^3)$
```
Comment (dongliu0426) [inline solutions/usaco-988.mdx:12]: why? N <= 1000 so N^3 = 1e9 operations D:

**Review by dongliu0426 — COMMENTED** [review]
oops, I'm late. @AAkashyap1 can you open a new PR to fix this?

File: solutions/usaco-988.mdx (hunk 10 above)
Comment (nathangong) [reply solutions/usaco-988.mdx:12]: Yeah wait shouldn't this be O(n^2)?

Comment (ryanchou-dev) [reply solutions/usaco-988.mdx:12]: > Yeah wait shouldn't this be O(n^2)?
yeah i think so

Comment (nathangong) [reply solutions/usaco-988.mdx:12]: lol is it possible to update and remerge a PR

Comment (dongliu0426) [reply solutions/usaco-988.mdx:12]: > Yeah wait shouldn't this be O(n^2)?

[his solution is O(n^3) :(](https://github.com/cpinitiative/usaco-guide/pull/1961#discussion_r673667982)

Comment (nathangong) [reply solutions/usaco-988.mdx:12]: Ah rip

**Pull request merged**